lib/mayaUsd/fileio/primUpdater.cpp

```diff
@@ -85,17 +85,21 @@ bool UsdMayaPrimUpdater::discardEdits()
     if (objectToDelete.isNull())
         return true;
 
+    // Nodes that are from a referenced file cannot be deleted
+    // as they live in a different file. They will go away once
+    // the reference is unloaded. Don't try to delete them here.
     MFnDependencyNode depNode(objectToDelete);
+    if (!depNode.isFromReferencedFile()) {
```

Comment:
As discussed, I think we should introduce pruning to the discardEdits traversal, like we did for mergeToUsd.  Otherwise, we'll needlessly traverse referenced nodes trying to discard their edits, when the edits to be discarded are in the Maya reference node itself.